bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java

```diff
@@ -149,15 +149,29 @@ public IProgressMonitor getProgressMonitor() {
 				: new NullProgressMonitor();
 
 		return new IProgressMonitor() {
+			private volatile boolean taskStarted;
```

Comment:
This is not sufficient for thread safety, so either it should be an atomicboolean using the compare and set operation below or a simple boolean, marking a value as volatile "just in case" does not help much and gives a false impression.

Reply:
IProgressMonitor itself is not designed to be threadsafe. look for example at the SubMonitor implementation which is inherently unsafe